doc/registry_api.md

```diff
@@ -0,0 +1,67 @@
+# Suggested API for registries
+
+```ruby
+# Returns the component matching the parameters from the registry or nil if
+# the component is not available. If multiple versions of the component is matching
+# the component with the highest version is returned.
+#
+# @param [String] component
+# @param [nil|String|Gem::Requirement|Gem::Version] version
+# @return [Diversity::Component|nil]
+# @alias get
```

Comment:
I think get more says that it will instantiate something, while [] indicates more of accessing what is already in a hash or array.